molgenis-security/src/main/java/org/molgenis/security/account/AccountController.java

```diff
@@ -150,18 +149,10 @@ public String changePasswordToken(
         throw new CaptchaException("invalid captcha answer");
       }
       User user = toUser(registerRequest);
-      String activationUri;
-      if (StringUtils.isEmpty(request.getHeader("X-Forwarded-Host"))) {
-        activationUri =
-            ServletUriComponentsBuilder.fromCurrentRequest()
-                .replacePath(URI + "/activate")
-                .build()
-                .toUriString();
-      } else {
-        String scheme = request.getHeader("X-Forwarded-Proto");
-        if (scheme == null) scheme = request.getScheme();
-        activationUri = scheme + "://" + request.getHeader("X-Forwarded-Host") + URI + "/activate";
-      }
+      String activationUri =
+          ServletUriComponentsBuilder.fromCurrentRequest()
```

Comment:
Might the ```ServletUriComponentsBuilderFactory``` bean be useful in cases such as these, e.g. to simplify testing?

Reply:
This factory currently sits in the `molgenis-api-data` module. Do you have a suggestion on where to put it? 

`molgenis-api-data` and  `molgenis-security` both share `molgenis-data` as a dependency but that doesn't sound like a nice fit to me.

Reply:
Using the factory introduces a lot of complexity in the tests so we'll leave it as is